### PR TITLE
Added resource file for version info

### LINKED ISSUE #16: Add versioning
NICE-TO-HAVE: It'll be great to add version in NppShell.dll

### PR DESCRIPTION
**Pull request by GurliGebis**
This adds a resource file for versioning the dll, with the metadata for version 8.5.2
This should fix #16

### CODE REVIEW

**Comment by GurliGebis** [comment]
@donho is this what you mean by versioning? 🙂

**Review by donho — CHANGES_REQUESTED** [review]
rc file is considered as binary. could you fix that please?

**Comment by GurliGebis** [comment]
@donho Sure, I'll try 🙂- right now it seems like git is insisting it is a binary file.
I'll see what I can do.

**Comment by GurliGebis** [comment]
@donho fixed - it was Visual Studio insisting on using UTF-16 with a BOM, causing git to insist it is a binary file.
Changed it to UTF-8, and now it works 🙂

File: NppShell.rc [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+#endif
+ FILEOS 0x40004L
+ FILETYPE 0x2L
+ FILESUBTYPE 0x0L
+BEGIN
+    BLOCK "StringFileInfo"
+    BEGIN
+        BLOCK "040904b0"
+        BEGIN
+            VALUE "CompanyName", "Don HO don.h@free.fr"
```
Comment (donho) [inline NppShell.rc:69]: Put your name and email (optional) please, since you are the author.

File: NppShell.rc [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+            VALUE "FileDescription", "Notepad++ Context Menu"
+            VALUE "FileVersion", "8.5.2.0"
+            VALUE "InternalName", "NppShell.dll"
+            VALUE "LegalCopyright", "Copyleft 1998-2023 by Don HO"
```
Comment (donho) [inline NppShell.rc:73]: Same here your name please.

File: NppShell.rc [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+ FILETYPE 0x2L
+ FILESUBTYPE 0x0L
+BEGIN
+    BLOCK "StringFileInfo"
+    BEGIN
+        BLOCK "040904b0"
+        BEGIN
+            VALUE "CompanyName", "Don HO don.h@free.fr"
+            VALUE "FileDescription", "Notepad++ Context Menu"
+            VALUE "FileVersion", "8.5.2.0"
```
Comment (donho) [inline NppShell.rc:71]: Though it is distributed with Notepad++, it's better to have its own version.
I would say v1.0 - what do you think?

File: NppShell.rc [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+#endif    // APSTUDIO_INVOKED
+
+
+/////////////////////////////////////////////////////////////////////////////
+//
+// Version
+//
+
+VS_VERSION_INFO VERSIONINFO
+ FILEVERSION 8,5,2,0
```
Comment (donho) [inline NppShell.rc:53]: Change version here also?

**Review by donho — CHANGES_REQUESTED** [review]
Please consider the suggestions

File: NppShell.rc (hunk 1 above)
Comment (GurliGebis) [reply NppShell.rc:69]: I will tomorrow 👍

File: NppShell.rc (hunk 2 above)
Comment (GurliGebis) [reply NppShell.rc:73]: Same as above, tomorrow 🙂

File: NppShell.rc (hunk 3 above)
Comment (GurliGebis) [reply NppShell.rc:71]: I agree, will fix tomorrow

File: NppShell.rc (hunk 4 above)
Comment (GurliGebis) [reply NppShell.rc:53]: Will do

**Comment by GurliGebis** [comment]
@donho also, can you include this in the 8.5.2 release? (Since we really should be having versioning info in the dll)

**Comment by donho** [comment]
> @donho also, can you include this in the 8.5.2 release? (Since we really should be having versioning info in the dll)

@GurliGebis 
Notepad++ v8.5.2 has been released 3 hours ago:
https://notepad-plus-plus.org/news/v852-released/

It (the commit of this PR) will go into v8.5.3 for sure :)

**Review by donho — APPROVED** [review]

**Pull request closed without merging**